ARGOS4VLDdemod/ByteSync.c

```diff
@@ -65,40 +65,40 @@ int FindSyncWords(unsigned char *bitStreamIn, double *bitStreamInTime, unsigned
             byte = 0;
             bitIdx = 0;
             frameByteIdx++;
-            if(frameByteIdx > 8)
+            if(frameByteIdx > 24)
```

Comment:
Remind me what this change does ?

Reply:
Ah, yes. It took me a while to find where the longer ARGOS 4 VLD Long messages were being truncated - and its there. That change just lets 24 bytes of data through instead of 8. Actually I think we only need 18 bytes, not 24. It might be nice to add the expected message length as an arg?

Reply:
yeah, good call. The best way would be to detect the end of the packet.... not so easy though!!

Is there a payload length byte we could read in to make a guess on??

Reply:
Yes, indeed:
The sync is 24 bits: 0xAC5353
The two bits after that define the length. If they are **0b00** then the total number of encoded data bits after that is 68. If the two length bits are **0b11** then 3*68 encoded data bits follow.
Note: this is only true for ARGOS 4 VLD. The other PTT formats use quite different length fields.
![image](https://user-images.githubusercontent.com/5690545/99145034-115f2d00-2663-11eb-83e3-57c3c2304fb1.png)

Reply:
Ah, but... The encoding process reduces the 68 bits down to 45. So, the true answer, in terms of how many bits to process, is:
- **0b00** indicates that **45** bits follow
- **0b11** indicates that **3*45** bits follow